File: backend/services/member-service/src/main/java/kr/lastdish/member/member/domain/Member.java
package kr.lastdish.member.member.domain;

import jakarta.persistence.*;
import java.time.LocalDateTime;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "members")
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Member {

  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  @Column(name = "member_id")
  private Long id;

  @Column(name = "user_name", nullable = false, unique = true, length = 50)
  private String userName;

  @Column(name = "password", nullable = false)
  private String password;

  @Column(name = "name", nullable = false, length = 50)
  private String name;

  @Column(name = "phone", nullable = false, length = 50)
  private String phone;

  @Column(name = "email", nullable = false, unique = true, length = 100)
  private String email;

  @Enumerated(EnumType.STRING)
  @Column(name = "role", nullable = false, length = 20)
  private Role role;

  @Column(name = "created_at", nullable = false, updatable = false)
  private LocalDateTime createdAt;

  @Column(name = "updated_at", nullable = false)
  private LocalDateTime updatedAt;

  @Column(name = "is_deleted", nullable = false)
  private Boolean isDeleted;

  @Column(name = "deleted_at")
  private LocalDateTime deletedAt;

  @Builder
  public Member(
      String userName, String password, String name, String phone, String email, Role role) {
    this.userName = userName;
    this.password = password;
    this.name = name;
    this.phone = phone;
    this.email = email;
    this.role = role;
    this.createdAt = LocalDateTime.now();
    this.updatedAt = LocalDateTime.now();
    this.isDeleted = false;
  }
}

File: frontend/lib/domain/model/member.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'member.freezed.dart';
part 'member.g.dart';

/// 내 회원 정보. 백엔드 `MemberProfileResponse`(`GET /members/me`)와 대응:
/// `{ id, userName, name, phone, email, role, createdAt, updatedAt }`.
///
/// 지금은 `id`(memberId)만 실제로 쓰인다 — Cart 조회(`GET /carts/members/{memberId}`)에
/// memberId가 필요한데 프론트가 로그인 응답만으로는 memberId를 몰라서, 로그인 후
/// 이 API를 한 번 호출해 얻어온다(`cart_repository_impl.dart` 참고). 나머지 필드는
/// 마이페이지(B12) 화면이 실제로 만들어지면 그때 쓰인다.
@freezed
class Member with _$Member {
  const factory Member({
    required int id,
    required String userName,
    required String name,
    required String phone,
    required String email,
    required String role,
    required String createdAt,
    required String updatedAt,
  }) = _Member;

  factory Member.fromJson(Map<String, Object?> json) => _$MemberFromJson(json);
}
